docs/testing-vk-servers.md

```diff
@@ -0,0 +1,201 @@
+	return len(runes)
+}
+```
+
+There are a few interesting things to point out in our server that are relevant to testing. The `attachRoutes()` function is reused later to setup our testing server in the same way it is used in `main()`. This is useful if you have a more complex routing setup.
+
+The other thing to note here is the `WCResponse` struct. We can reuse it directly to make writing tests a breeze.
+
+```go
+// server.go
+func attachRoutes(server *vk.Server) {
+	api := vk.Group("/api/v1").Before(createWordcountMiddleware)
+	api.POST("/wc", handlePost)
+
+	server.AddGroup(api)
+}
+
+func createWordcountMiddleware(r *http.Request, ctx *vk.Ctx) error {
+	text, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return err
+	}
+
+	wc := Wordcount(string(text))
+	ctx.Set(wordcountCtxKey, wc)
+
+	return nil
+}
+
+type WCResponse struct {
+	Words      int `json:"words"`
+	Lines      int `json:"lines"`
+	Characters int `json:"chars"`
+}
+
+func NewWCResponse(wc Wordcount) *WCResponse {
+	return &WCResponse{
+		Words:      wc.Words(),
+		Lines:      wc.Lines(),
+		Characters: wc.Characters(),
+	}
+}
+
+func handlePost(r *http.Request, ctx *vk.Ctx) (interface{}, error) {
+	wc := ctx.Get(wordcountCtxKey).(Wordcount)
+
+	return vk.R(http.StatusOK, NewWCResponse(wc)), nil
+}
+
+func main() {
+	server := vk.New(vk.UseAppName("wordcount"), vk.UseHTTPPort(9090))
+	attachRoutes(server)
+
+	if err := server.Start(); err != nil {
+		log.Fatal(err)
+	}
+}
+```
+
+As a quick check, let's run our wordcount server.
+
+```
+cd docs/examples/wordcount
+go run .
+```
+
+You should see the default logger output:
+```
+{"log_message":"(W) configured to use HTTP with no TLS","timestamp":"2021-08-16T10:45:46.273974908-04:00","level":2}
+{"log_message":"(I) starting wordcount ...","timestamp":"2021-08-16T10:45:46.274132326-04:00","level":3}
+{"log_message":"(I) serving on :9090","timestamp":"2021-08-16T10:45:46.274144389-04:00","level":3}
+```
+
+Let's send it a request. In a separate terminal run:
+```
+curl http://localhost:9090/api/v1/wc -d "Hello, word count"
+```
+
+Which should produce:
+```json
+{"words":3,"lines":1,"chars":17}
+```
+
+## Writing Tests
+Now that our server seems to be running as expected, let's write tests with the `vtest` package.
+
+The `vtest` package is part of Vektor and can be found in at [github.com/suborbital/vektor/tree/main/vtest](https://github.com/suborbital/vektor/tree/main/vtest).
+
+Tests written with `vtest` use the usual Go testing idioms. If you're not familiar with the `testing` package, you can [read about it here](https://golang.org/doc/tutorial/add-a-test) as part of the official Go tutorial.  
+
+Here is a complete test function for a Vektor server. Let's break it down.
+```go
+func TestWordcount(t *testing.T) {
+	server := vk.New(vk.UseAppName("wordcount_testing"))
+	attachRoutes(server)
+
+	vt := vtest.New(server)
+
+    body := strings.NewReader("There's a starman waiting in the sky\nHe'd like to come and meet us")
```

Comment:
Indentation

Reply:
For some reason that line uses spaces instead of tabs.